gocourse02/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"fmt"
+	"math/rand/v2"
+)
+
+/*
+Написати програму “Зоопарк”. Звіри повтікали (більше трьох штук), наглядач повинен їх зібрати.
+Кожна сутність (наглядач, звір, клітка, …) представляється окремою структурою (zookeeper, animal, cage, …).
+Треба використати щонайменше: структури, вказівник, nil, будування, конструктор. Додати тваринам можливість
+розмножуватись. Програма має демонструвати свою роботу через вивід в stdout.
+*/
+
+const escapedAnimalsCount = 15 // Кількість звірів які повтікали
+
+type ZooKeeper struct {
+	Name          string
+	AnimalsFound  int
+	AnimalsCaught int
+}
+
+type Catcher interface {
+	FindEscapedAnimal() bool                // Спроба знайти звіра
+	AttemptCatchAnimal(*Cage, *Animal) bool // Сброба зловити звіра в клітку
+}
+
+type Animal struct {
+	ID     int
+	Name   string
+	Gender string
+	Weight int
+}
+
+type Cage struct {
+	MaxSeats      int
+	MaxWeight     int
+	CurrentSeats  int
+	CurrentWeight int
+	Animal        []Animal
+}
+
+func NewAnimal(id, weight int) *Animal {
+	fakeData := NewFake()
+	return &Animal{ID: id + 1, Name: fakeData.Name, Gender: fakeData.Gender, Weight: weight}
+}
+
+func NewZooKeeper(name string) *ZooKeeper {
+	return &ZooKeeper{name, 0, 0}
+}
+
+func NewCage(animNumber int) *Cage {
+	numberSeats := RandInt(1, animNumber+1)    // Дамо звірям шанс
+	maxWeight := numberSeats * RandInt(10, 50) // Максимальная вага одного звіра в зоопарку 50
+	animals := make([]Animal, 0)
```

Comment:
Нам відомо кількість звірів в клітці? Якщо так, то можна:
```suggestion
	animals := make([]Animal, 0, animNumber)
```

Reply:
Зробив так animals := make([]Animal, 0, numberSeats+1). Максимальная кількість звірів яка може поміститися в клітку +1=child